rotkehlchen/db/history_events.py

```diff
@@ -355,36 +368,35 @@ def get_history_events(
         TODO: To not query all columns with all joins for all cases, we perhaps can
         peek on the entry type of the filter and adjust the SELECT fields accordingly?
         """
-        free_query_group_by = ''
-        free_query_count = ''
         base_prefix = 'SELECT '
-        type_idx = 0
+        base_suffix = f'{HISTORY_BASE_ENTRY_FIELDS}, {EVM_EVENT_FIELDS}, {ETH_STAKING_EVENT_FIELDS} {ALL_EVENTS_DATA_JOIN}'  # noqa: E501
+        bindings = []
+        type_idx = 1 if group_by_event_ids else 0
         special_free_query = False
-        if group_by_event_ids is True:
-            if has_premium:
+        free_query_group_by = maybe_filter_ignore_asset(filter_query)
+
+        if has_premium is True:
+            if group_by_event_ids is True:
                 base_prefix += 'COUNT(*), '
-            else:  # a bit ugly conditions to keep limit at groups for free users
-                free_query_count = 'COUNT(*), '
-                for fil in filter_query.filters:
-                    if isinstance(fil, DBIgnoredAssetsFilter):
-                        # Also don't count spam asset transactions in the limit
-                        free_query_group_by = 'WHERE (asset IS NULL OR asset NOT IN (SELECT value FROM multisettings WHERE name="ignored_asset")) '  # noqa: E501
-                        break
+            base_query = f'{base_prefix} {base_suffix}'
+        else:
+            if group_by_event_ids is True:
+                special_free_query = True  # a bit ugly conditions to keep limit at groups for free users  # noqa: E501
                 free_query_group_by += 'GROUP BY event_identifier'
-                special_free_query = True
-
-            type_idx = 1
+                base_query = f'{base_prefix} * FROM (SELECT COUNT(*), {base_suffix} '
+                base_query += f'{free_query_group_by} ORDER BY timestamp DESC, sequence_index ASC LIMIT ?) '  # noqa: E501
```

Comment:
same if you can avoid having the `+=`operator is better. Reason is that you have to mutate and store twice the string and that is harder to read since you are just adding to the end but mentally you have to read what is in the variable. Is easier to read it in just in once

```suggestion
                base_query = (
                    f'{base_prefix} * FROM (SELECT COUNT(*), {base_suffix} '
                    f'{free_query_group_by} ORDER BY timestamp DESC, sequence_index ASC LIMIT ?) '
                )
```